R/utils-get_code_dependency.R

```diff
@@ -0,0 +1,498 @@
+# get_code_dependency ----
+
+#' Return the lines of code (with side-effects) needed to reproduce the object
+#'
+#' @details This function assumes that object relationships are established using the `<-`, `=`, or `->` assignment
+#' operators. It does not support other object creation methods like `assign` or `<<-`, nor non-standard-evaluation
+#' methods. To specify relationships between side-effects and objects, you can use the comment tag
+#' `# @linksto object_name` at the end of a line where the side-effect occurs.
+#'
+#' @param code An `expression` with `srcref` attribute or a `character` with the code.
+#' @param names A `character(n)` with object names.
+#'
+#' @return `character` vector of elements of `code` calls that were required to build the side-effects and
+#' influencing objects having and impact on the `object`
+#'
+#' @keywords internal
+get_code_dependency <- function(code, names) {
+  assert_classes(code, names)
+
+  if (is_empty(code)) return(code)
+
+  code <- assert_code(code) # Turns code into expression with srcref attribute.
+  pd <- utils::getParseData(code)
+  assert_names(names, pd)
+
+  calls_pd <- extract_calls(pd)
+
+  graph <- code_graph(calls_pd)
+  indexes <- unlist(lapply(names, function(x) graph_parser(x, graph)))
+  as.character(code[unique(indexes)])
+}
+
+#' Group the result of `utils::getParseData()` into separate calls
+#' @param pd (`data.frame`) A result of `utils::getParseData()`.
```

Comment:
```suggestion
#' @param pd (`data.frame`) A result of [utils::getParseData()] limited to three columns (`"token"`, `"text"`, `"id"`)
```

Reply:
I think the result of this function is limited, but not the input parameter. Maybe we can move this to `@return`/`Value` section : )

Reply:
true, my mistake

Reply:
no worries, c05f27f382473b3abb89a11b61b5ac581ed5e843